### PR TITLE
fix: conventional-commits can match more characters in () identifier

### PR DESCRIPTION
**Pull request by edulop91**
so we can do things like `feat(FOOBAR-123): do the thing`

### CODE REVIEW

**Review by jakeyheath — APPROVED** [review]

File: .github/actions/conventional-commits/action.yml [hunk 1]
```diff
@@ -1,20 +1,78 @@
 name: Conventional Commits Checking
 description: "Validates that a pull request title is compatible with Conventional Commits."
+inputs:
+  enforce_identifier_prefixes:
+    description: "List of JIRA (or similar) project identifiers to enforce. eg: `feat(FOO-123):...` or `feat(BAR-123):...` then pass in FOO\nBAR"
```
Comment (jakeyheath) [inline .github/actions/conventional-commits/action.yml:5]: I will add that this description is kind of confusing. I'm not really sure what it means.

Comment (jakeyheath) [reply .github/actions/conventional-commits/action.yml:5]: Can you provide an example of what format you are expecting here. Like is it a string of characters that you are allowing or is it a regexp or something else.

Comment (edulop91) [reply .github/actions/conventional-commits/action.yml:5]: https://github.com/chanzuckerberg/github-actions/pull/90/files#diff-e27c2676c8ee63a108d6d91680cbeca9fdce68ff977cc4914b149f7e40b065f0R16

File: .github/actions/conventional-commits/action.yml [hunk 2]
```diff
@@ -1,20 +1,78 @@
...
+      with:
+        node-version: "18"
+        check-latest: true
+    - run: npm install csv-parse
+      shell: bash
     - uses: actions/github-script@v5
       with:
         script: |
-          const validator = /^(chore|feat|fix|revert|docs|style)(\([a-z ]+\))?(!)?: (.)+$/
+          const {parse} = require('csv-parse/sync');
```
Comment (jakeyheath) [inline .github/actions/conventional-commits/action.yml:19]: If we are going to be using a library, is there just a node library that checks this already? all this regex stuff seems very error prone, but you know my feelings about regexp.

Comment (jakeyheath) [reply .github/actions/conventional-commits/action.yml:19]: like this? https://github.com/conventional-changelog/commitlint

File: .github/actions/conventional-commits/action.yml (hunk 1 above)
Comment (jakeyheath) [reply .github/actions/conventional-commits/action.yml:5]: ya I guess that example doesn't help me understand the newline character mentioned in this description

File: .github/workflows/conventional_commits_title.yml [hunk 3]
```diff
@@ -13,3 +13,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/conventional-commits
+      - uses: ./.github/actions/conventional-commits
+        with:
+          enforce_identifier_prefixes: |
+            CCIE
```
Comment (edulop91) [inline .github/workflows/conventional_commits_title.yml:19]:
```suggestion
            CCIE
            CID
```

File: .github/workflows/conventional_commits_title.yml [hunk 4]
```diff
@@ -13,3 +13,8 @@ jobs:
...
+            CID
```
Comment (edulop91) [inline .github/workflows/conventional_commits_title.yml:20]:
```suggestion
            CDI
```

File: .github/actions/conventional-commits/action.yml (hunk 2 above)
Comment (edulop91) [reply .github/actions/conventional-commits/action.yml:19]: you're right

**Comment by jakeyheath** [comment]
stale

**Pull request closed without merging**